### PR TITLE
Reorganize sharing design doc introduction

### PR DESCRIPTION
**Pull request by Gara64**
This is meant to better introduce the core sharing concepts

### CODE REVIEW

File: docs/sharing-design.md [hunk 1]
```diff
@@ -1,35 +1,61 @@
...
+- The shared data is duplicated among members: it is both on the sharer's cozy
+  (the owner), and on the recipients' cozy. This is a core difference compared to
+  a federated sharing such as the one in
+  [NextCloud](https://docs.nextcloud.com/server/latest/user_manual/files/federated_cloud_sharing.html).
+- The [CouchDB replication
+  protocol](https://docs.couchdb.org/en/stable/replication/protocol.html) is used
+  to synchronize the documents. It means that the documents will be same on the
+  owner side and on the recipients side ("symmetric sharing").
+- The applications say what is shared, the cozy stack synchronizes that. The
+  stack does the low-level stuff and gives primitives to the applications. The
```
Comment (Crash--) [inline docs/sharing-design.md:19]:
```suggestion
  stack does the low-level work and gives primitives to the applications. The
```

Comment (Crash--) [inline docs/sharing-design.md:19]: ` gives primitives to the applications` which are?

Comment (Gara64) [reply docs/sharing-design.md]: I assume it is the sharing routes and the html for permissions, discovery, etc. @nono?

Comment (nono) [reply docs/sharing-design.md]: Yes, it's correct

**Pull request merged**